### ISSUE TITLE
Unable to finetune Honeybee/Llama2 with matsci-nlp data.

### ISSUE THREAD

**Issue by wiseyy** [BangLab-UdeM-Mila/NLP4MatSci-HoneyBee#3]
Hi, while finetuning Llama2 with matsci-nlp data provided in the repo, I faced the following issues.

I finetuned the model using the code given in uniform_finetune.py.

1. The default cutoff length for the prompt/generated text is 512. But there were plenty of examples that exceeded this limit, so I had to increase it to either 1024 or 2048. Was the original model finetuned with 512 itself? If yes, how?
2. The finetuning is done in a Seq2Seq manner and not for text generation. Why is that? Even after finetuning, I am not getting the output that is easy to evaluate according to the scripts given to the repo. The output is not a single word (named entity in case of NER). 

<img width="854" alt="Screenshot 2024-01-19 at 10 32 49 AM" src="https://github.com/BangLab-UdeM-Mila/NLP4MatSci-HoneyBee/assets/77568830/f35f7a8c-b79f-4fda-bb5a-7ee80afbe9af">

Can you please clarify the above points, and let us know if you performed additional post processing before evaluating Honeybee's output on the matsci-nlp dataset?

**Comment by yusonghust** [comment]
1. No, the original model finetuned with 2048, so you can change the hyper-params
2. The finetuning is done for text generation, you can check the code carefully. The Llama is a decoder-only language model, it can only predict the next token based on previous tokens.

**Comment by wiseyy** [comment]
<img width="1084" alt="Screenshot 2024-01-21 at 7 52 23 AM" src="https://github.com/BangLab-UdeM-Mila/NLP4MatSci-HoneyBee/assets/77568830/e9ce43e1-62d2-4b0d-8f3c-ec7d53eab863">

1. Why do you use Seq2Seq Collator here and not DataCollatorForLanguageModeling? I think it is for computing the loss only on the answer, given the instruction and the input, and the behaviour will be the same (should be, atleast) if you change the attention mask to attend to the tokens that follow after "###Response: ". Can you please confirm this?

2. Do you use the prompts as given in uniform_finetune.py to finetune over matsci-nlp.py as well?

Thank you for your help!

**Comment by yusonghust** [comment]
1、please refer to https://github.com/tloen/alpaca-lora/issues/412
2、yes and it is finetuned under low-resource setting
